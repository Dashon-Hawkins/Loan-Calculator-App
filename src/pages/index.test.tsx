import React from 'react';
import renderer from 'react-test-renderer';
import IndexPage from './';

describe('Index page', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<IndexPage />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
