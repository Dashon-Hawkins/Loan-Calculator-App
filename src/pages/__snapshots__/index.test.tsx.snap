// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index page renders correctly 1`] = `
<div
  className="css-1mp5wj"
>
  <main
    className="css-19ugk9f"
  >
    <div
      className="MuiBox-root MuiBox-root-5"
    >
      <h1
        className="css-14jsf90"
      >
        Кредитный калькулятор
      </h1>
      <div
        className="css-seia6q"
      >
        <form
          onReset={[Function]}
          onSubmit={[Function]}
        >
          <div
            className="css-s4sngx"
          >
            <div
              className="MuiBox-root MuiBox-root-6"
            >
              <div
                className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
              >
                <label
                  className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                  data-shrink={false}
                >
                  Сумма кредита
                </label>
                <div
                  className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
                  onClick={[Function]}
                >
                  <input
                    aria-invalid={false}
                    className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
                    disabled={false}
                    name="amount"
                    onAnimationStart={[Function]}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onMouseUp={[Function]}
                    required={false}
                    type="text"
                    value=""
                    variant="outlined"
                  />
                  <div
                    className="MuiInputAdornment-root MuiInputAdornment-positionEnd"
                  >
                    <button
                      aria-label="toggle term currency"
                      className="MuiButtonBase-root MuiIconButton-root makeStyles-inputButton-3"
                      disabled={false}
                      onBlur={[Function]}
                      onClick={[Function]}
                      onDragLeave={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      tabIndex={0}
                      type="button"
                    >
                      <span
                        className="MuiIconButton-label"
                      >
                        ₽
                      </span>
                      <span
                        className="MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                  <fieldset
                    aria-hidden={true}
                    className="PrivateNotchedOutline-root-84 MuiOutlinedInput-notchedOutline"
                    style={
                      Object {
                        "paddingLeft": 8,
                      }
                    }
                  >
                    <legend
                      className="PrivateNotchedOutline-legend-85"
                      style={
                        Object {
                          "width": 0.01,
                        }
                      }
                    >
                      <span
                        dangerouslySetInnerHTML={
                          Object {
                            "__html": "&#8203;",
                          }
                        }
                      />
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
            <div
              className="MuiBox-root MuiBox-root-86"
            >
              <div
                className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
              >
                <label
                  className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                  data-shrink={false}
                >
                  Срок кредита
                </label>
                <div
                  className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
                  onClick={[Function]}
                >
                  <input
                    aria-invalid={false}
                    className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
                    disabled={false}
                    name="term"
                    onAnimationStart={[Function]}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    required={false}
                    type="text"
                    value=""
                  />
                  <div
                    className="MuiInputAdornment-root MuiInputAdornment-positionEnd"
                  >
                    <p
                      className="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
                    >
                      мес
                    </p>
                  </div>
                  <fieldset
                    aria-hidden={true}
                    className="PrivateNotchedOutline-root-84 MuiOutlinedInput-notchedOutline"
                    style={
                      Object {
                        "paddingLeft": 8,
                      }
                    }
                  >
                    <legend
                      className="PrivateNotchedOutline-legend-85"
                      style={
                        Object {
                          "width": 0.01,
                        }
                      }
                    >
                      <span
                        dangerouslySetInnerHTML={
                          Object {
                            "__html": "&#8203;",
                          }
                        }
                      />
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
          </div>
          <div
            className="css-s4sngx"
          >
            <div
              className="MuiBox-root MuiBox-root-117"
            >
              <div
                className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
              >
                <label
                  className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined"
                  data-shrink={true}
                >
                  Начало выплат
                </label>
                <div
                  className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
                  onClick={[Function]}
                >
                  <input
                    aria-invalid={false}
                    className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
                    disabled={false}
                    name="start"
                    onAnimationStart={[Function]}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    required={false}
                    type="text"
                    value="04.10.2019"
                  />
                  <div
                    className="MuiInputAdornment-root MuiInputAdornment-positionEnd"
                  >
                    <button
                      className="MuiButtonBase-root MuiIconButton-root"
                      disabled={false}
                      onBlur={[Function]}
                      onClick={[Function]}
                      onDragLeave={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      tabIndex={0}
                      type="button"
                    >
                      <span
                        className="MuiIconButton-label"
                      >
                        <svg
                          aria-hidden="true"
                          className="MuiSvgIcon-root"
                          focusable="false"
                          role="presentation"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                          />
                          <path
                            d="M0 0h24v24H0z"
                            fill="none"
                          />
                        </svg>
                      </span>
                      <span
                        className="MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                  <fieldset
                    aria-hidden={true}
                    className="PrivateNotchedOutline-root-84 MuiOutlinedInput-notchedOutline"
                    style={
                      Object {
                        "paddingLeft": 8,
                      }
                    }
                  >
                    <legend
                      className="PrivateNotchedOutline-legend-85"
                      style={
                        Object {
                          "width": 0,
                        }
                      }
                    >
                      <span
                        dangerouslySetInnerHTML={
                          Object {
                            "__html": "&#8203;",
                          }
                        }
                      />
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
            <div
              className="MuiBox-root MuiBox-root-146"
            >
              <div
                className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
              >
                <label
                  className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                  data-shrink={false}
                >
                  Процентная ставка
                </label>
                <div
                  className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
                  onClick={[Function]}
                >
                  <input
                    aria-invalid={false}
                    className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
                    disabled={false}
                    name="rate"
                    onAnimationStart={[Function]}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    required={false}
                    type="text"
                    value=""
                  />
                  <div
                    className="MuiInputAdornment-root MuiInputAdornment-positionEnd"
                  >
                    <p
                      className="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
                    >
                      %
                    </p>
                  </div>
                  <fieldset
                    aria-hidden={true}
                    className="PrivateNotchedOutline-root-84 MuiOutlinedInput-notchedOutline"
                    style={
                      Object {
                        "paddingLeft": 8,
                      }
                    }
                  >
                    <legend
                      className="PrivateNotchedOutline-legend-85"
                      style={
                        Object {
                          "width": 0.01,
                        }
                      }
                    >
                      <span
                        dangerouslySetInnerHTML={
                          Object {
                            "__html": "&#8203;",
                          }
                        }
                      />
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
          </div>
          <div
            className="css-s4sngx"
          >
            <div
              className="MuiFormControl-root makeStyles-formControl-1"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
                data-shrink={true}
                htmlFor="outlined-age-simple"
              >
                Порядок погашения
              </label>
              <div
                className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
                onClick={[Function]}
              >
                <div
                  aria-haspopup="listbox"
                  className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSelect MuiOutlinedInput-inputSelect"
                  id="select-procedure"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  role="button"
                  tabIndex={0}
                >
                  Аннуитетный
                </div>
                <input
                  name="procedure"
                  onAnimationStart={[Function]}
                  type="hidden"
                  value="A"
                />
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
                <fieldset
                  aria-hidden={true}
                  className="PrivateNotchedOutline-root-84 MuiOutlinedInput-notchedOutline"
                  style={
                    Object {
                      "paddingLeft": 8,
                    }
                  }
                >
                  <legend
                    className="PrivateNotchedOutline-legend-85"
                    style={
                      Object {
                        "width": 0,
                      }
                    }
                  >
                    <span
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "&#8203;",
                        }
                      }
                    />
                  </legend>
                </fieldset>
              </div>
            </div>
            <div
              className="MuiBox-root MuiBox-root-159"
            >
              <label
                className="MuiFormControlLabel-root"
              >
                <span
                  aria-disabled={false}
                  className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-172 MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-checked-173 Mui-checked MuiIconButton-colorPrimary"
                  onBlur={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={null}
                >
                  <span
                    className="MuiIconButton-label"
                  >
                    <input
                      checked={true}
                      className="PrivateSwitchBase-input-175"
                      data-indeterminate={false}
                      disabled={false}
                      onChange={[Function]}
                      type="checkbox"
                      value="round"
                    />
                    <svg
                      aria-hidden="true"
                      className="MuiSvgIcon-root"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                      />
                    </svg>
                  </span>
                  <span
                    className="MuiTouchRipple-root"
                  />
                </span>
                <span
                  className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                >
                  Округлять
                </span>
              </label>
            </div>
          </div>
          <div
            className="css-1agno4y"
          >
            <button
              className="MuiButtonBase-root MuiButton-root makeStyles-button-2 MuiButton-contained"
              disabled={false}
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="submit"
            >
              <span
                className="MuiButton-label"
              >
                Рассчитать
              </span>
              <span
                className="MuiTouchRipple-root"
              />
            </button>
          </div>
        </form>
      </div>
    </div>
  </main>
  <aside
    className="css-vls53n"
  >
    <div
      className="MuiBox-root MuiBox-root-195"
    >
      <span
        className="css-1bq2kn9"
      >
        <svg
          fill="currentColor"
          height="1.5em"
          size="1.5em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 640 512"
          width="1.5em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M640 51.2l-.3 12.2c-28.1.8-45 15.8-55.8 40.3-25 57.8-103.3 240-155.3 358.6H415l-81.9-193.1c-32.5 63.6-68.3 130-99.2 193.1-.3.3-15 0-15-.3C172 352.3 122.8 243.4 75.8 133.4 64.4 106.7 26.4 63.4.2 63.7c0-3.1-.3-10-.3-14.2h161.9v13.9c-19.2 1.1-52.8 13.3-43.3 34.2 21.9 49.7 103.6 240.3 125.6 288.6 15-29.7 57.8-109.2 75.3-142.8-13.9-28.3-58.6-133.9-72.8-160-9.7-17.8-36.1-19.4-55.8-19.7V49.8l142.5.3v13.1c-19.4.6-38.1 7.8-29.4 26.1 18.9 40 30.6 68.1 48.1 104.7 5.6-10.8 34.7-69.4 48.1-100.8 8.9-20.6-3.9-28.6-38.6-29.4.3-3.6 0-10.3.3-13.6 44.4-.3 111.1-.3 123.1-.6v13.6c-22.5.8-45.8 12.8-58.1 31.7l-59.2 122.8c6.4 16.1 63.3 142.8 69.2 156.7L559.2 91.8c-8.6-23.1-36.4-28.1-47.2-28.3V49.6l127.8 1.1.2.5z"
          />
        </svg>
      </span>
      <div
        className="MuiBox-root MuiBox-root-196"
      >
        <b>
          Аннуитетный платёж
          .
        </b>
         
        Равный по сумме ежемесячный платёж по кредиту, который включает в себя сумму начисленных процентов за кредит и сумму основного долга. Расчёт аннуитетного платежа в банках производится по несколько разным формулам. Поэтому даже при одинаковой процентной ставке размер аннуитетного платежа может различаться у разных банков.
        <br />
        <br />
        <b>
          Дифференцированный платёж
          .
        </b>
         
        Ежемесячный платёж по кредиту, уменьшающийся к концу срока кредитования и состоит из выплачиваемой постоянной доли основного долга и процентов на невыплаченный остаток кредита.
      </div>
    </div>
  </aside>
</div>
`;
